aws-greengrassv2-componentversion/src/main/java/software/amazon/greengrassv2/componentversion/Translator.java

```diff
@@ -123,14 +123,7 @@ static DeleteComponentRequest translateToDeleteRequest(final ResourceModel model
   static ListComponentVersionsRequest translateToListRequest(final ResourceHandlerRequest<ResourceModel> request,
                                                              final String nextToken) {
     return ListComponentVersionsRequest.builder()
-            .arn(Arn.builder()
-                    .partition(request.getAwsPartition())
-                    .region(request.getRegion())
-                    .accountId(request.getAwsAccountId())
-                    .service("greengrass")
-                    .resource("components:" + request.getDesiredResourceState().getComponentName())
-                    .build()
-                    .toString())
+            .arn(request.getDesiredResourceState().getArn())
```

Comment:
Discussed offline. In list use case, arn is component arn instead of component version arn. According to the guidance, List use cases are independent from the typical CFN use case so we do not need additional field for separation.